backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,159 @@
+      "src/test/java/com/google/sps/images/data/dogRunningOnBeach.jpg", "jpg");
+    emptyLabelAnnotations = new ArrayList<>();
+
+    labelAnnotations = new ArrayList<>();
+
+    EntityAnnotation entityAnnotationMock = mock(EntityAnnotation.class);
+    // mock the behavior of stock service to return the value of various stocks
+    when(entityAnnotationMock.getDescription()).thenReturn("DescriptionOne");
+
+    labelAnnotations.add(entityAnnotationMock);
+    labelAnnotations.add(entityAnnotationMock);
+  }
+
+  /**
+   * Returns an image as a byte array from the local reference of the image.
+   *
+   * @param reference the local reference of the image.
+   * @param fileType the file extension of the image.
+   * @return the byte array representation of the image.
+   */
+  private static byte[] getBytesFromImageReference(String reference, String fileType) throws IOException {
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      throw exception;
+    }
+
+    return bos.toByteArray();
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is null.
+   *  labelAnnotations is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(nullRawImageData, emptyLabelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is empty.
+   *  labelAnnotations is non-empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorEmptyImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(emptyRawImageData, labelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is non-empty image data.
+   *  labelAnnotations is non-empty.
+  */
+  @Test
+  public void constructorRawImageData() throws IllegalArgumentException {
+    AnnotatedImage actual = new AnnotatedImage(rawImageData, labelAnnotations);
+    assertTrue(Arrays.equals(rawImageData, actual.getRawImageData()));
+    assertEquals(labelAnnotations, actual.getLabelAnnotations());
+  }
+
+  /**
+   * Tests paritions on getLabelDescriptions();
+   *  labelAnnotations is empty.
+  */
+  @Test
+  public void getLabelDescriptionsEmptyLabelAnnotations() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(rawImageData, emptyLabelAnnotations);
+    List<String> actual = annotatedImageActual.getLabelDescriptions();
```

Comment:
Nit: `actual` -> `actualLabelDescriptions` for increased descriptiveness (When I see `actual` I wonder "actual what?")

Reply:
Sounds good! I'll update this and keep it in mind for future testing.